src/common/settings_objects.cpp

```diff
@@ -277,27 +278,43 @@ namespace PowerToysSettings
         return L"RESOURCE ID NOT FOUND: " + std::to_wstring(resource_id);
     }
 
-    PowerToyValues::PowerToyValues(std::wstring_view powertoy_name)
+    PowerToyValues::PowerToyValues(std::wstring_view powertoy_name, std::wstring_view powertoy_key)
     {
-        _name = powertoy_name;
+        _key = powertoy_key;
         set_version();
         m_json.SetNamedValue(L"name", json::value(powertoy_name));
         m_json.SetNamedValue(L"properties", json::JsonObject{});
     }
 
-    PowerToyValues PowerToyValues::from_json_string(std::wstring_view json)
+    PowerToyValues PowerToyValues::from_json_string(std::wstring_view json, std::wstring_view powertoy_key)
     {
         PowerToyValues result = PowerToyValues();
+        json::JsonObject jsonObject = json::JsonValue::Parse(json).GetObjectW();
+        if (!jsonObject.HasKey(L"name"))
+        {
+            throw winrt::hresult_error(E_NOT_SET, L"name field not set");
+        }
+
+        if (!jsonObject.HasKey(L"properties"))
+        {
+            throw winrt::hresult_error(E_NOT_SET, L"properties field not set");
+        }
+
+        if (!jsonObject.HasKey(L"version"))
+        {
+            throw winrt::hresult_error(E_NOT_SET, L"version field not set");
+        }
```

Comment:
@somil55 the the PowerPreview tests are failing because these exceptions get thrown.
It fails on this test, since Properties and version is not passed in the argument https://github.com/microsoft/PowerToys/blob/36dd29c05631a54b7c327f52b9bcbcb9ae954e5c/src/modules/previewpane/powerpreviewTest/FileExplorerPreviewSettingsTest.cpp#L94-L107